File: leetCode/901.js

var StockSpanner = function() {
  this.spans = []
  this.stockPrices = []
};

/**
 * @param {number} price
 * @return {number}
 */
StockSpanner.prototype.next = function(price) {
  let count = 1
  while (this.stockPrices.length > 0 && price >= this.stockPrices[this.stockPrices.length - 1]) {
    this.stockPrices.pop()
    count += this.spans.pop()
  }
  this.stockPrices.push(price)
  this.spans.push(count)
  return count
};

var obj = new StockSpanner()
var param_1 = obj.next(10)

/**
 * Your StockSpanner object will be instantiated and called as such:
 * var obj = new StockSpanner()
 * var param_1 = obj.next(price)
 */

File: leetCode/901.py
class StockSpanner:

    def __init__(self):
        self.prices = []
        self.spans = []

    def next(self, price: int) -> int:
        span = 1
        index = len(self.spans) - 1
        while index >= 0 and price >= self.prices[index]:
            span += self.spans[index]
            index -= self.spans[index]
        self.spans.append(span)
        self.prices.append(price)
        return span



# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)